go/vt/mysqlctl/mysqld.go

```diff
@@ -157,18 +152,16 @@ func (mysqld *Mysqld) RunMysqlUpgrade() error {
 		return nil
 	}
 
-	// run the program, if it fails, we fail
+	// Run the program, if it fails, we fail.  Note in this
+	// moment, mysqld is running with no grant tables on the local
+	// socket only, so this doesn't need any password for root.
 	args := []string{
 		// --defaults-file=* must be the first arg.
 		"--defaults-file=" + mysqld.config.path,
 		"--socket", mysqld.config.SocketFile,
-		"--user", mysqld.dba.Uname,
+		"--user", "root",
```

Comment:
It definitely doesn't have to exist. With --skip-grant-tables MySQL doesn't even know which users exist or not. So if something failed for you it wasn't related to the existence of the user.